components/st-microelectronics/stm-api.stanza

```diff
@@ -125,137 +133,181 @@ defn parse (my-ref: Ref) -> String :
+  connect-battery(stm, bat-net, VDD)
 
-public defn connect-reset (stm:JITXObject) :
+public defn connect-battery (stm:JITXObject) :
+  connect-battery(stm, VBAT, VDD)
+
+public defn connect-reference (stm:JITXObject, ref-net:ReferencePinNames, vdd-net:PwrPinNames) :
+  inside pcb-module :
+  ; TODO Add reference option to API, currently assumes none.
+    val r-net = get-pin-list(stm, ref-net)
+    if not empty?(r-net) :
+      val p-pin = get-pin-list(stm, vdd-net)
+      if not empty?(p-pin) :
+        net (r-net p-pin[0])
+
+public defn connect-reset (stm:JITXObject, reset-net:ResetPinNames, vdd-net:PwrPinNames, gnd-net:GndPinNames, reset-pullup:Double, reset-cap:Double) :
   inside pcb-module :
     pin reset
-    if find-pins?(stm, "NRST") :
-      net (stm.NRST reset)
-      val power-pin = power-pins(stm, VDD)[0]
-      val gnd-pin = gnd-pins(stm, VSS)[0]
-      res-strap(power-pin, stm.NRST, 10.0e3)
-      cap-strap(power-pin, stm.NRST, 10.0e-9)
-
-      ; Look for power-on reset pins
-      if find-pins?(stm, "NPOR") :
-        val v-range = recommended-voltage(property(power-pin.power-pin))
-        if contains?(v-range, 1.8):
+    val reset-pin-list = get-pin-list(stm, reset-net)
+    if not empty?(reset-pin-list) :
+      val power-pins = power-pins-list(stm, VDD)
+      for p in reset-pin-list do :
+        net (p reset)
+        if not empty?(power-pins) :
+          match(reset-pullup) :
+            (r: Double)       : res-strap(power-pins[0], p, r)
+          match(reset-cap) :
+            (c: Double)     : cap-strap(power-pins[0], p, c)
+
+public defn connect-reset (stm:JITXObject, reset-net:ResetPinNames, vdd-net:PwrPinNames, gnd-net:GndPinNames) :
+  val settings = Settings(DEFAULT-SETTINGS)
+  connect-reset(stm reset-net vdd-net gnd-net settings[`reset-pullup] settings[`reset-cap])
+
+public defn connect-reset (stm:JITXObject, reset-net:ResetPinNames, reset-pullup:Double, reset-cap:Double) :
+  connect-reset(stm reset-net VDD VSS reset-pullup reset-cap)
+
+public defn connect-reset (stm:JITXObject, reset-net:ResetPinNames) :
+  val settings = Settings(DEFAULT-SETTINGS)
+  connect-reset(stm reset-net VDD VSS settings[`reset-pullup] settings[`reset-cap])
+
+public defn connect-reset (stm:JITXObject, reset-pullup:Double, reset-cap:Double) :
+  connect-reset(stm NRST VDD VSS reset-pullup reset-cap)
+
+public defn connect-reset (stm:JITXObject) :
+  val settings = Settings(DEFAULT-SETTINGS)
+  connect-reset(stm NRST VDD VSS settings[`reset-pullup] settings[`reset-cap])
+
+public defn connect-por (stm:JITXObject, por-net:ResetPinNames, vdd-net:PwrPinNames, gnd-net:GndPinNames) :
+  inside pcb-module :
+    pin por
+    val por-pin = get-pin-list(stm, por-net)
+    if not empty?(por-pin) :
+      val power-pins = power-pins-list(stm, vdd-net)
+      val gnd-pins = gnd-pins-list(stm, gnd-net)
+      val v-range = recommended-voltage(property(power-pins[0].power-pin))
+      if not empty?(power-pins) and not empty?(gnd-pins) :
```

Comment:
Perhaps `v-range` should be calculated after knowing `power-pins` is not empty?

Reply:
Yes, good feedback. Changed.